roles/variables/vars/main.yml

```diff
@@ -181,8 +182,8 @@ kafka_controller_properties:
     properties:
       zookeeper.set.acl: 'true'
   migration_broker_listener:
-    enabled: "{{kafka_controller_sasl_protocol=='plain' and kraft_migration|bool }}"
-    properties: "{{  {'broker_listener': kafka_broker_listeners[kafka_broker_inter_broker_listener_name]} | confluent.platform.listener_properties(kafka_broker_ssl_enabled, fips_enabled, kafka_broker_ssl_mutual_auth_enabled, kafka_broker_sasl_protocol,
+    enabled: "{{ sasl_protocol=='plain' and kraft_migration|bool }}"
+    properties: "{{  {'broker_listener': kafka_broker_listeners[kafka_broker_inter_broker_listener_name]} | confluent.platform.listener_properties(ssl_enabled, fips_enabled, ssl_mutual_auth_enabled, sasl_protocol,
```

Comment:
Can you please explain why are we making this change

Reply:
although this is officially out of the scope of cp-ansible, I was trying out a scenario when broker has sasl enabled and controlled doesn't. In that case also, we need this listener config for the communication between controller and broker since controller now acts as client and broker as server. In greenfield kraft, controller was always the server so this communication was not required

Reply:
Got it. What if the `sasl_enabled` is not enabled but `kafka_controller_sasl_protocol` is set to plain. This might be a corner case, but lets handle the situations.

Reply:
tested this case, it works

Reply:
I tried out this edge case, our playbook currently does not handle different controller-broker and broker-broker communication. Therefore, to migrate in such a way the customer would need to use override configs